### PR TITLE
[no-release-notes] deprecating setJoinScopeLen

### PR DESCRIPTION
**Pull request by JCOR11599**
With new name resolution and better indexing, we can deprecate this old rule with much simpler code.

### CODE REVIEW

File: sql/analyzer/fix_exec_indexes.go [hunk 1]
```diff
@@ -388,6 +387,24 @@ func (s *idxScope) finalizeSelf(n sql.Node) (sql.Node, error) {
 		if nc, ok := ret.(sql.CheckConstraintNode); ok && s.checks != nil {
 			ret = nc.WithChecks(s.checks)
 		}
+		if jn, ok := ret.(*plan.JoinNode); ok {
+			if len(s.parentScopes) == 0 {
```
Comment (max-hoffman) [inline sql/analyzer/fix_exec_indexes.go:391]: if there were >1 parent, would we summate all the parent offsets?

Comment (max-hoffman) [reply sql/analyzer/fix_exec_indexes.go:391]: what happened to the HashLookup special case? does that one not matter?

**Review by max-hoffman — APPROVED** [review]
I like the concise point refactor. I also think we are short of general correctness without accounting for arbitrary scope nesting. That doesn't need to be in this PR, but I don't think it'd be hard to comment out some of the skipped join/lateral join tests that stress scope nesting and see if we can fix a couple bugs in the process.

File: sql/analyzer/fix_exec_indexes.go (hunk 1 above)
Comment (JCOR11599) [reply sql/analyzer/fix_exec_indexes.go:391]: as far as I can tell, we no longer have to prevent StripRowNodes from being added over HashLookups

**Pull request merged**